client/components/title-format-editor/index.jsx

```diff
@@ -388,14 +397,16 @@ export class TitleFormatEditor extends Component {
 const mapStateToProps = ( state, ownProps ) => {
 	const site = getSelectedSite( state );
 	const { translate } = ownProps;
+	const userLocale = getCurrentUserLocale( state );
+	const formattedDate = moment().locale( userLocale ).format( 'MMMM YYYY' );
```

Comment:
> I think this should depend on current site's language, instead of user locale, in order to match the actual result that will be generated by SEO tools.

Makes sense, I'll see if that's possible!